src/store/index.js

```diff
@@ -34,25 +38,76 @@ export default new Vuex.Store({
       state.collections = await state.connection.db.listCollections().toArray()
     },
     async find({ state }, query = {}) {
-      state.totalNumberOfResults = null
+      if (state.collectionRecords[state.collectionName]) {
+        const query_serialization = JSON.stringify(query)
+
+        if (state.collectionRecords[state.collectionName].query === query_serialization) {
```

Comment:
could you explain what this logic is doing here? it suggests to me that i won't be able to run the same query multiple times consecutively on the same collection, is that right?

Reply:
My assumption is if the query has already run it's not necessary to re-run it again. If do you have another suggestion I would like to hear it.